iib/workers/tasks/build.py

```diff
@@ -833,20 +923,94 @@ def _adjust_operator_manifests(manifests_path):
         for pullspec in operator_csv.get_pullspecs():
             found_pullspecs.add(pullspec)
 
+    conf = get_worker_config()
+    registry_replacements = (
+        conf['iib_organization_customizations']
+        .get(organization, {})
+        .get('registry_replacements', {})
+    )
+
     # Resolve pull specs to container image digests
     replacement_pullspecs = {}
     for pullspec in found_pullspecs:
-        # Skip images that are already pinned
-        if ':' not in ImageName.parse(pullspec).tag:
-            replacement_pullspecs[pullspec] = ImageName.parse(_get_resolved_image(pullspec))
+        replacement_needed = False
+        if ':' in ImageName.parse(pullspec).tag:
+            resolved_image = ImageName.parse(pullspec)
```

Comment:
I wonder if it makes sense to always use `_get_resolved_image` to ensure the pull spec is valid?

Reply:
I avoided it because `skopeo inspect` without `--raw` is quite expensive but I added it based on your recommendation in a separate commit.